### PR TITLE
FDA better start guess for FFT

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
The plotting in FDA (muon) looked like it was not working. This was because the workspace selection in FFT was just the first in the list. This PR makes it choose the same group/pair as the home tab, if the current selection is not available. 

**To test:**
1 Open frequency domain analysis
2 Load some EMU data
3 Go to phase table tab and make sure all of the combo boxes are populated
4 Go to FFT tab
5 The worksapce and imaginary workspace selection will be for the long pair (long will be in the name)
6 Change the selection for the workspace and imaginary workspace
7 Go to the grouping tab and create another pair
8 Check that the new pair has appeared in the group/pair selection on the home tab
9 Check the selection in the FFT tab has not changed
10 Go to the grouping tab and delete one of the groups/pairs you have selected
11 Go to FFT
12 The deleted selection should have defaulted back to long. 

Fixes #2563. 

In the release notes
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by Mantid-Matthew** [comment]
I get the RunTimeError below when trying to plot fft spectrums created in the transform tab. I create a fft transform of a group and then try and plot it with the Frequency FFT All option.

```
Traceback (most recent call last):
  File "/home/matthew/Documents/Mantid/mantid/scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py", line 111, in handle_plot_type_changed
    self.plot_standard_workspaces()
  File "/home/matthew/Documents/Mantid/mantid/scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_presenter.py", line 134, in plot_standard_workspaces
    self._model.plot(workspace_list, self.get_plot_title(), self.get_domain(), self._force_redraw, self.context.window_title)
  File "/home/matthew/Documents/Mantid/mantid/scripts/Muon/GUI/Common/home_plot_widget/home_plot_widget_model.py", line 103, in plot
    errors=True)
  File "/home/matthew/Documents/Mantid/mantid/qt/python/mantidqt/plotting/functions.py", line 60, in inner_func
    return func(workspaces, *args, **kwargs)
  File "/home/matthew/Documents/Mantid/mantid/qt/python/mantidqt/plotting/functions.py", line 225, in plot
    plot_fn(ws, **plot_kwargs)
  File "/home/matthew/Documents/Mantid/mantid/Framework/PythonInterface/mantid/plots/__init__.py", line 64, in wrapper
    func_value = func(self, *args, **kwargs)
  File "/home/matthew/Documents/Mantid/mantid/Framework/PythonInterface/mantid/plots/__init__.py", line 711, in errorbar
    _data_update, spec_num, is_normalized)
  File "/home/matthew/Documents/Mantid/mantid/Framework/PythonInterface/mantid/plots/__init__.py", line 315, in track_workspace_artist
    spec_num))
  File "/home/matthew/Documents/Mantid/mantid/Framework/PythonInterface/mantid/plots/__init__.py", line 97, in __init__
    self.workspace_index = self._get_workspace_index()
  File "/home/matthew/Documents/Mantid/mantid/Framework/PythonInterface/mantid/plots/__init__.py", line 105, in _get_workspace_index
    return ads.retrieve(self.workspace_name).getIndexFromSpectrumNumber(self.spec_num)
RuntimeError: Could not find spectrum number in any spectrum.
```

**Comment by AnthonyLim23** [comment]
unrelated failure

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py [hunk 1]
```diff
@@ -59,14 +59,37 @@ def deactivate(self):
...
         final_options = self.load.get_workspace_names_for_FFT_analysis(self.view.use_raw_data)
 
+        # update view
         self.view.addItems(final_options)
         self.view.removeRe('PhaseQuad')
         self.removePhaseFromIM(final_options)
 
-        self.view.workspace = name
+        # make intelligent guess of what user wants
+        group_pair_context = self.load.group_pair_context[self.load.group_pair_context.selected]
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py]: I think this name is misleading as unless I'm mistaken this is just the currently selected group/pair object?

File: scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py [hunk 2]
```diff
@@ -59,14 +59,37 @@ def deactivate(self):
...
+        Re_name_to_use = None
+        Im_name_to_use = None
+        default_name = None
+        # will need to check this exists before using it
+        if group_pair_context:
```
Comment (Mantid-Matthew) [inline scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py]: Is this needed when do we expect the relevant group/pair object to nor exist? Also if this check fails will line 85 below not produce a RuntimeError as it tries to index a None object?

Comment (AnthonyLim23) [reply scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py]: if `group_pair_context` does not exist and the current name is not in the list of new names we go to line 84. In this case `default_name` is None so we do not enter the if statement.

**Review by Matthew-Andrew — APPROVED** [review]
This works for me :+1:

**Comment by martyngigg** [comment]
The build failure was unrelated.

**Pull request merged**